ls1intum/Artemis
# Add explanation to modelling exercise

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin)

### Description
Added explanation for modeling exercise creation, modeling exercise example submission and modeling exercise submission

### Steps for Testing
Note: Example Submission is not being listed is not related to this pull request.

1. Create a modeling exercise give example solution explanation, create example submission with explanation
2. Log in as student and create a submission with explanation
3. Log in as tutor and check if you can see if the user can see example solution explanation and explanation for submissions in assessment
4. Assessing the explanation will be added with followup pull request
5. Follow same steps for modeling exercise in an exam
6. Check if explanation is missing anywhere modeling exercise submission related 

### Test Coverage
modeling-exam-submission.component.ts 96.97%
modeling-exam-summary.component.ts 100%
modeling-explanation-editor.component.ts 100%
modeling-assessment.component.ts  78.63%

### Screenshots
#### Modeling Exercise Creation with Example Solution Explanation
![out-19](https://user-images.githubusercontent.com/16825065/102770476-84f00a80-4384-11eb-8192-e2c2357adbe0.gif)
#### Example Submission with Explanation for Modeling Exercise
![out-20](https://user-images.githubusercontent.com/16825065/102770481-87526480-4384-11eb-8fc9-35c05fb94902.gif)
#### Modeling Exercise Submission with Explanation
![out-21](https://user-images.githubusercontent.com/16825065/102770923-3858ff00-4385-11eb-9b49-d7d08ae169e1.gif)
#### Modeling Exercise Assessment: Example Solution with Explanation
![out-22_1](https://user-images.githubusercontent.com/16825065/102770683-d5fffe80-4384-11eb-83c2-64f1c7e59585.gif)
#### Modeling Exercise Assessment: Submission with Explanation
![out-23](https://user-images.githubusercontent.com/16825065/102770691-d8faef00-4384-11eb-9b90-751d6ba46879.gif)
#### Modeling Exercise in Exam with Explanation
![out-24](https://user-images.githubusercontent.com/16825065/102770695-da2c1c00-4384-11eb-9dcc-bb8587972439.gif)
#### Team Exercise with Explanation
![out-26](https://user-images.githubusercontent.com/16825065/102770706-ddbfa300-4384-11eb-9ffc-06bd3201f528.gif)
